### PR TITLE
ASoC: max98373: Added fix for amp global ON/OFF stability.

### PR DESCRIPTION
**Pull request by ryans-lee**
This PR includes 3 patches to improve stability of AMP ON/OFF.

In some case like chip overheating and current over drawn, amp could be shutdown by itself.

1) Since amp on/off register(0x20FF) was nonvolatile, amp ON event has been ignored due to regmap cache sync.
0x20FF register was changed as volatile to avoid this issue.

2) 30ms delay after amp ON/OFF was added based on datasheet.

3) Auto-restart functions that recovers amp from abnormal state caused by overheating and overcurrent was enabled by default.
Additional controls have been added to on/off auto-restart function.

### CODE REVIEW

File: sound/soc/codecs/max98373-i2c.c [hunk 1]
```diff
@@ -446,6 +446,7 @@ static bool max98373_volatile_reg(struct device *dev, unsigned int reg)
 	case MAX98373_R2054_MEAS_ADC_PVDD_CH_READBACK:
 	case MAX98373_R2055_MEAS_ADC_THERM_CH_READBACK:
 	case MAX98373_R20B6_BDE_CUR_STATE_READBACK:
+	case MAX98373_R20FF_GLOBAL_SHDN:
```
Comment (plbossart) [inline sound/soc/codecs/max98373-i2c.c:449]: the commit message is a bit unclear.

Is the description that on a regmap sync we don't restore a value that was modified in hardware? I don't specifically get what the 'Amp ON command has been ignored' would mean.

also the indents should be fixed, this looks like a gitk copy/paste.

File: sound/soc/codecs/max98373.c [hunk 2]
```diff
@@ -204,6 +204,15 @@ SOC_SINGLE("Ramp Up Switch", MAX98373_R203F_AMP_DSP_CFG,
...
 	MAX98373_AMP_DSP_CFG_RMP_DN_SHIFT, 1, 0),
+/* Speaker Amplifier Overcurrent Automatic Restart Enable */
+SOC_SINGLE("OVC Autorestart Switch", MAX98373_R20FE_DEVICE_AUTO_RESTART_CFG,
+	MAX98373_OVC_AUTORESTART_SHIFT, 1, 0),
+/* Thermal Shutdown Automatic Restart Enable */
+SOC_SINGLE("THERM Autorestart Switch", MAX98373_R20FE_DEVICE_AUTO_RESTART_CFG,
+	MAX98373_THERM_AUTORESTART_SHIFT, 1, 0),
+/* Clock Monitor Automatic Restart Enable */
+SOC_SINGLE("CMON Autorestart Switch", MAX98373_R20FE_DEVICE_AUTO_RESTART_CFG,
+	MAX98373_CMON_AUTORESTART_SHIFT, 1, 0),
```
Comment (plbossart) [inline sound/soc/codecs/max98373.c:215]: typo in commit message:

3 new controls were added

**Review by plbossart — CHANGES_REQUESTED** [review]
looks good @ryans-lee but commit messages can be polished a bit.

File: sound/soc/codecs/max98373-i2c.c (hunk 1 above)
Comment (ryans-lee) [reply sound/soc/codecs/max98373-i2c.c:449]: Thanks. Shall modify the commit message and indentation as well.

File: sound/soc/codecs/max98373.c (hunk 2 above)
Comment (ryans-lee) [reply sound/soc/codecs/max98373.c:215]: Shall fix this.

File: sound/soc/codecs/max98373-i2c.c (hunk 1 above)
Comment (plbossart) [reply sound/soc/codecs/max98373-i2c.c:449]: typo ignnored -> ignored.

**Review by plbossart — CHANGES_REQUESTED** [review]
minor typo in commit message.

If this helps you can send this directly to Mark Brown, we don't have a lot of added value here.

File: sound/soc/codecs/max98373-i2c.c (hunk 1 above)
Comment (ryans-lee) [reply sound/soc/codecs/max98373-i2c.c:449]: Fixed typo. Thanks.

**Comment by ryans-lee** [comment]
> minor typo in commit message.
> 
> If this helps you can send this directly to Mark Brown, we don't have a lot of added value here.

Thanks for your help. I sent patches directly to Mark Brown and it have been applied to https://git.kernel.org/pub/scm/linux/kernel/git/broonie/sound.git

**Comment by plbossart** [comment]
thanks @ryans-lee

**Pull request closed without merging**